src/Elasticsearch/Metadata/Resource/Factory/ElasticsearchProviderResourceMetadataCollectionFactory.php

```diff
@@ -26,8 +26,11 @@
 
 final class ElasticsearchProviderResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
 {
-    public function __construct(private readonly Client $client, private readonly ResourceMetadataCollectionFactoryInterface $decorated)
+    public function __construct(private readonly Client $client, private readonly ResourceMetadataCollectionFactoryInterface $decorated, private readonly bool $triggerDeprecation = true)
```

Comment:
I don't understand how this works, shouldn't we deprecate inside the `create` method instead? And only if the `elasticsearch` flag is used?

Reply:
This service will always be instantiated, even if the user did not set `$elasticsearch`. In that case we should not log a deprecation.